tst/EnergyPlus/unit/ConstructionInternalSource.unit.cc

```diff
@@ -91,3 +91,32 @@ TEST_F(EnergyPlusFixture, ConstructionInternalSource)
 
     EXPECT_NEAR(0.1524, state->dataConstruction->Construct(1).ThicknessPerpend, 0.0001);
 }
+
+TEST_F(EnergyPlusFixture, ConstructionInternalSourceEmptyField)
+{
+
+    // test whether empty "Two-Dimensional Temperature Calculation Position" causes reading errors
+    std::string const idf_objects = delimited_string({
+        "	ConstructionProperty:InternalHeatSource,	",
+        "	Radiant Source,          !- Name",
+        "	Slab Floor with Radiant, !- Construction Name",
+        "	4,                       !- Source Present After Layer Number",
+        "	4,                       !- Temperature Calculation Requested After Layer Number",
+        "	2,                       !- Dimensions for the CTF Calculation",
+        "	0.3048,                  !- Tube Spacing {m}",
+        "	;                        !- Two-Dimensional Temperature Calculation Position",
+        "	Construction,	",
+        "	Slab Floor with Radiant, !- Name",
+        "	CONCRETE - DRIED SAND AND GRAVEL 4 IN,  !- Outside Layer",
+        "	INS - EXPANDED EXT POLYSTYRENE R12 2 IN,  !- Layer 2",
+        "	GYP1,                    !- Layer 3",
+        "	GYP2,                    !- Layer 4",
+        "	FINISH FLOORING - TILE 1 / 16 IN;  !- Layer 5",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    bool errorsFound(false);
+
+    GetConstructData(*state, errorsFound);
```

Comment:
Are you supposed to `ASSERT_FALSE(errorsFound)` here?  Or is simply calling it without it failing a sufficient test?

Reply:
> Are you supposed to `ASSERT_FALSE(errorsFound)` here? Or is simply calling it without it failing a sufficient test?

If that's the entire unit test, then yes, I would add `EXPECT_FALSE(errorsFound); `
But I would also add an expect for the value of the 2-d input field that's the whole point of this.

p.s. What's the difference, if any, between `ASSERT_FALSE` and `EXPECT_FALSE`?

Reply:
AFAIK, EXPECT should let the function continue running, possibly catching additional EXPECT failures.  ASSERT will fail immediately.

Reply:
So errorsFound is actually coming back as true.  Something seems wrong.  Perhaps it is the material names are invalid?  Maybe it's OK that errorsFound comes back true, but we can still check to make sure the CP:IHS object was still set up properly on its own?  I'm not sure but I'm holding here for now and moving to other PRs until this is nailed down.  Thanks for the confirmation @mjwitte

Reply:
@yujiex this comment is still hanging, so I'm going to hold from any more review here.  Pull develop in for another round of testing once this is resolved.

Reply:
I agree this unit test is incomplete. Add material objects for the materials referenced in the construction object (in this test and in `ConstructionInternalSource` above), so that errorsfound will be false. Check the value of the input field in question in both unit tests.